kudu/src/main/java/org/apache/apex/malhar/kudu/ApexKuduConnection.java

```diff
@@ -128,8 +131,20 @@ public void setKuduClient(KuduClient kuduClient)
     this.kuduClient = kuduClient;
   }
 
-  public static class ApexKuduConnectionBuilder
+  public ApexKuduConnectionBuilder getBuilderForThisConnection()
+  {
+    return builderForThisConnection;
+  }
+
+  public void setBuilderForThisConnection(ApexKuduConnectionBuilder builderForThisConnection)
   {
+    this.builderForThisConnection = builderForThisConnection;
+  }
+
+  public static class ApexKuduConnectionBuilder implements Serializable
```

Comment:
Any specific reason this is marked Serializable? By default kryo is used to serialize the state and it doesn't need classes to implement Serializable.

Reply:
Was more thinking of Java serialisability support. I can remove it but it would not harm either if java serialisation needs to be considered for all derived classes.